Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -534,9 +534,16 @@ internal async Task<string> SelectChild(TreeNode treeNode)
                 {
                     return cs.Child;
                 }
-                if ((bool)await this.EvaluateDynamicProperty(cs.ShouldSelect, typeof(bool)).ConfigureAwait(false))
+                try
                 {
-                    return cs.Child;
+                    if ((bool)await EvaluateDynamicProperty(cs.ShouldSelect, typeof(bool)).ConfigureAwait(false))
+                    {
+                        return cs.Child;
+                    }
+                }
+                catch (Exception e)
+                {
+                    throw new Exception($"An error occurred while evaluating 'ShouldSelect' expression: '{cs.ShouldSelect}' in the ChildSelector with child '{cs.Child}': '{e.Message}'.", e);
```

Comment:
Hmm.. My one concern is that this change would change the behavior when we DO hit an EvaluateDynamicPropertyException when calling EvaluateDynamicProperty. We would throw an Exception instead, which leads to a "Failed" Status instead of "Failed_EvaluateDynamicProperty". Perhaps if we broke the statement into 2 lines. One line to call EvaluateDynamicProperty. Then inside a try/catch, cast to bool and return.